apps/web/src/pages/[document].tsx

```diff
@@ -126,18 +130,21 @@ const DocumentPage: NextPage<Props> = () => {
       onMutate: ({ page }) => {
         trackEvent("Page Save Clicked");
 
-        const previousPage = queryClient.getQueryData<DeserializedPage>(
-          DESERIALIZED_PAGE_QUERY_KEY
-        );
+        const previousPage =
+          queryClient.getQueryData<DeserializedPageQueryData>(
+            DESERIALIZED_PAGE_QUERY_KEY
+          );
 
         if (previousPage) {
-          queryClient.setQueryData<DeserializedPage>(
+          const optimisticallyUpdatedPage: DeserializedPage = {
+            ...previousPage.page,
+            title: page.title,
+            data: page.content,
+          };
+
+          queryClient.setQueryData<DeserializedPageQueryData>(
             DESERIALIZED_PAGE_QUERY_KEY,
-            {
-              ...previousPage,
-              title: page.title,
-              data: page.content,
-            }
+            { ...previousPage, page: optimisticallyUpdatedPage }
```

Comment:
Nice catch! 🚀
Maybe just rename `previousPage` to `previous` as it contains `key` as well?